### PR TITLE
Restore pruning functionality

### PR DESCRIPTION
**Pull request by SupremoUGH**
- Restores pruning functionality. Goes together with https://github.com/Manta-Network/manta-rs/pull/353
- Also adds UTXO consolidation. Goes together with https://github.com/Manta-Network/manta-rs/pull/355
- Adds the to public optimization in https://github.com/Manta-Network/manta-rs/pull/356.
- Adds the sync protocol optimization in https://github.com/Manta-Network/manta-rs/pull/357
- Add Api `estimateTransferPostsCount`
- Add Api `consolidateTransferSend`

---

Before we can merge this PR, please make sure that all the following items have been checked off:

- [x] Linked to an issue with discussion and accepted design OR have an explanation in the PR that describes this work.
- [x] Added **one** line describing your change in [`CHANGELOG.md`](https://github.com/manta-network/sdk/blob/main/CHANGELOG.md) and added the appropriate `L-` label to the PR.
- [ ] Re-reviewed `Files changed` in the GitHub PR explorer.

### CODE REVIEW

File: manta-js/package/src/wallet/crate/src/lib.rs [hunk 1]
```diff
@@ -1785,6 +1820,82 @@ impl Wallet {
...
+    pub fn post_consolidation(&self, request: JsValue, network: Network) -> Promise {
+        self.with_async(
+            |this| Box::pin(this.post_consolidation(ConsolidationPrerequest::new(request).into())),
+            network,
+        )
+    }
+
+    /// Returns the estimated number of [`TransferPost`]s necessary to execute the `transaction`.
+    #[inline]
+    pub fn estimate_transferposts(&self, transaction: Transaction, network: Network) -> usize {
```
Comment (ferrell-code) [inline manta-js/package/src/wallet/crate/src/lib.rs:1892]:
```suggestion
    pub fn estimate_transfer_posts(&self, transaction: Transaction, network: Network) -> usize {
```

nit, fine as is too

File: manta-js/package/src/pallets/MantaPayWallet.ts [hunk 2]
```diff
@@ -135,6 +135,124 @@ export default class MantaPayWallet
...
+        const originUtxoList = utxoList.map((item: UtxoInfo) => {
+          return {
+            asset: {
+              id: bnToU8a(bnToBn(item.asset.id), { bitLength: 256 }),
+              value: nToBigInt(item.asset.value),
+            },
+            identifier: {
+              is_transparent: item.identifier.is_transparent,
+              utxo_commitment_randomness: bnToU8a(
+                bnToBn(item.identifier.utxo_commitment_randomness),
```
Comment (ferrell-code) [inline manta-js/package/src/pallets/MantaPayWallet.ts:174]: Shouldn't the encoding here be the same as above or maybe I am missing why format is different

Comment (ferrell-code) [reply manta-js/package/src/pallets/MantaPayWallet.ts:174]: Line 151 has this block instead

```rust
            utxo_commitment_randomness: u8aToBn(
              item.identifier.utxo_commitment_randomness,
            ).toString(),
```

Comment (DanielZhangReal) [reply manta-js/package/src/pallets/MantaPayWallet.ts:174]: `getAllUtxoList` is to convert the u8a format into a wallet format that is easier to use

`consolidateTransactionBuild` is the list in `getAllUtxoList` that the user selects, and then revert the format to what wasm expects.

**Review by ferrell-code — APPROVED** [review]

**Review by ghzlatarev — APPROVED** [review]

**Pull request merged**